### PR TITLE
[KOGITO-5612] BDD tests: Migrate performance tests to Hyperfoil

### PR DESCRIPTION
**Pull request by sutaakar**
Signed-off-by: Karel Suta <ksuta@redhat.com>

https://issues.redhat.com/browse/KOGITO-5612

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've ran `make before-pr` and everything is working accordingly
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

File: test/features/kogito_service_performance.feature [hunk 1]
```diff
@@ -9,53 +9,145 @@ Feature: Kogito Service Performance
...
+      | usersPerSec | maxSessions |
+      | 100         | 110         |
 
 #####
 
   @quarkus
   @persistence
   @infinispan
-  Scenario Outline: Quarkus Kogito Service Performance with native <native>, with persistence and with requests <requests>
+  Scenario Outline: Quarkus Kogito Service Performance with native false, with persistence, usersPerSec <usersPerSec> and maxSessions <maxSessions>
```
Comment (radtriste) [inline test/features/kogito_service_performance.feature:150]:
```suggestion
  Scenario Outline: Quarkus Kogito Service Performance with native disabled, with persistence, usersPerSec <usersPerSec> and maxSessions <maxSessions>
```

File: test/features/kogito_service_performance.feature [hunk 2]
```diff
@@ -65,84 +157,230 @@ Feature: Kogito Service Performance
...
+      | usersPerSec | maxSessions |
+      | 100         | 110         |
+
+#####
+
+  @quarkus
+  @native
+  @persistence
+  @infinispan
+  Scenario Outline: Quarkus Kogito Service Performance with native true, with persistence, usersPerSec <usersPerSec> and maxSessions <maxSessions>
```
Comment (radtriste) [inline test/features/kogito_service_performance.feature:239]:
```suggestion
  Scenario Outline: Quarkus Kogito Service Performance with native true, with enabled, usersPerSec <usersPerSec> and maxSessions <maxSessions>
```

**Review by radtriste — COMMENTED** [review]
This looks good to me. just small comments to be in sync in scenario titles with other features

**Comment by sutaakar** [comment]
@radtriste Fixed

**Review by radtriste — APPROVED** [review]

**Review by MarianMacik — APPROVED** [review]
Looks good, nice work! We can just discuss the comments I have here.

File: test/features/kogito_service_performance.feature [hunk 3]
```diff
@@ -9,53 +9,145 @@ Feature: Kogito Service Performance
...
+    And Store benchmark statistics of Hyperfoil instance "hf-controller" as "quarkus-hotspot.json"
 
     Examples:
-      | native   | requests |
-      | disabled | 40000    |
-      | disabled | 80000    |
-      # | disabled | 160000   |
-      # | disabled | 320000   |
+      | usersPerSec | maxSessions |
+      | 100         | 110         |
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply test/features/kogito_service_performance.feature:86]: Yes, can be left to another PR.

File: test/features/kogito_service_performance.feature [hunk 4]
```diff
@@ -9,53 +9,145 @@ Feature: Kogito Service Performance
...
+                      "approver" : "john",
+                      "order" : {
+                        "orderNumber" : "12345",
+                        "shipped" : false
+                      }
+                    }
       """
 
-    Then Service "process-quarkus-example" contains <requests> instances of process with name "orders" within 1 minutes
-    And Service "process-quarkus-example" contains <requests> instances of process with name "orderItems" within 1 minutes
```
[the remark this answers is not in the document]
Comment (sutaakar) [reply test/features/kogito_service_performance.feature:34]: I can leave it there, commented, as currently we don't know how many requests we will make with current benchmarks.

**Comment by sutaakar** [comment]
@radtriste Can be merged

**Pull request merged**